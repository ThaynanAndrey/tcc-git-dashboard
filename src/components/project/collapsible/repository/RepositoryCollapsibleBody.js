import React from 'react';
import ProjectRepositoryList from '../../../repository/ProjectRepositoryList';

/**
 * Stateless component with the function of presenting a Collapsible Body to Repositories.
 * 
 * @author Thaynan Nunes
 */
const RepositoryCollapsibleBody = ({ idProject }) => (
    <div className="collapsible-body body-collapsible-project">
        <ProjectRepositoryList idProject={idProject} />
    </div>
);

export default RepositoryCollapsibleBody;