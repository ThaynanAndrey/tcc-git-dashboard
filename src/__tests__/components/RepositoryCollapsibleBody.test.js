import React from 'react';
import { shallow, configure } from "enzyme";
import toJson from 'enzyme-to-json';

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import RepositoryCollapsibleBody from '../../components/project/collapsible/repository/RepositoryCollapsibleBody';

describe('Test RepositoryCollapsibleBody', () => {
  it('renders correctly a component to show RepositoryCollapsibleBody', () => {
    const idProject = "123";
    const tree = shallow(
        <RepositoryCollapsibleBody idProject={idProject}/>
    );

    expect(toJson(tree)).toMatchSnapshot();
    expect(tree.find(".collapsible-body.body-collapsible-project").exists()).toBeTruthy();
  });
});