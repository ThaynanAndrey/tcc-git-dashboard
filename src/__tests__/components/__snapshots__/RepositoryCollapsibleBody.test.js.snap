// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test RepositoryCollapsibleBody renders correctly a component to show RepositoryCollapsibleBody 1`] = `
<div
  className="collapsible-body body-collapsible-project"
>
  <AuthenticatedComponent
    idProject="123"
  />
</div>
`;
